allennlp/tests/models/basic_classifier_test.py

```diff
@@ -1,16 +1,30 @@
-from allennlp.common.testing import ModelTestCase, AllenNlpTestCase
+import numpy
+
+from allennlp.common.testing import ModelTestCase
 
 
 class TestBasicClassifier(ModelTestCase):
+    def setUp(self):
+        super(TestBasicClassifier, self).setUp()
+        self.set_up_model(self.FIXTURES_ROOT / 'basic_classifier' / 'experiment_seq2vec.jsonnet',
+                          self.FIXTURES_ROOT / 'data' / 'text_classification_json' / 'imdb_corpus.jsonl')
+
+    def test_forward_pass_runs_correctly(self):
+        training_tensors = self.dataset.as_tensor_dict()
+        output_dict = self.model(**training_tensors)
+        output_dict = self.model.decode(output_dict)
+        assert 'label' in output_dict.keys()
+        probs = output_dict['probs'][0].data.numpy()
+        numpy.testing.assert_almost_equal(numpy.sum(probs, -1), numpy.array([1]))
 
     def test_seq2vec_clf_can_train_save_and_load(self):
-        data_directory = AllenNlpTestCase.FIXTURES_ROOT / "data" / "text_classification_json"
-        self.set_up_model(AllenNlpTestCase.FIXTURES_ROOT / 'basic_classifier' / 'experiment_seq2vec.jsonnet',
-                          data_directory / "imdb_corpus.jsonl")
+        self.set_up_model(self.FIXTURES_ROOT / 'basic_classifier' / 'experiment_seq2vec.jsonnet',
```

Comment:
the `setUp` method sets up things you will need for the whole class for the duration of the test, so actually, you can remove these duplicated calls to `self.set_up_model` now you have pulled it out into the `setUp` method.

Reply:
It seems that `pytest` runs test in alphabetical order, so the tests are run in this order:
- `test_forward_pass_runs_correctly`
- `test_seq2seq_clf_can_train_save_and_load`
- `test_seq2vec_clf_can_train_save_and_load`

Right now the 1st and 3rd tests above require the same model, so I added the repeated `set_up_model` calls everywhere. I'll change it so that tests 1 and 2 above use the same model, so there will still be 1 `set_up_model` call for the 3rd test.

Reply:
Sorry, I missed that they are different fixtures being set up. How you have it is fine.